adapters/concert/concert.go

```diff
@@ -0,0 +1,164 @@
+package concert
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/errortypes"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+	"net/http"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+const adapterVersion = "1.0.0"
+
+// Builder builds a new instance of the Concert adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	bidder := &adapter{
+		endpoint: config.Endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	bidderImpExt, err := getBidderExt(request.Imp[0])
+	if err != nil {
+		return nil, []error{fmt.Errorf("get bidder ext: %v", err)}
+	}
+
+	requestJSON, err := json.Marshal(request)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	var requestMap map[string]interface{}
+	err = json.Unmarshal(requestJSON, &requestMap)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	if requestMap["ext"] == nil {
+		requestMap["ext"] = make(map[string]interface{})
+	}
+	requestMap["ext"].(map[string]interface{})["adapterVersion"] = adapterVersion
+	requestMap["ext"].(map[string]interface{})["partnerId"] = bidderImpExt.PartnerId
+
+	requestJSON, err = json.Marshal(requestMap)
+	if err != nil {
+		return nil, []error{err}
+	}
+
+	requestData := &adapters.RequestData{
```

Comment:
Update `MakeRequests` to return `ImpIDs` as mentioned below :-
 
https://github.com/prebid/prebid-server/blob/763551963dd3208dde6f9bf35b31ef76cc83814f/adapters/pubmatic/pubmatic.go#L175-L182

Reply:
Done! Thank you for sharing the pubmatic example.